dbms/src/Storages/StorageFactory.cpp

```diff
@@ -558,14 +558,14 @@ StoragePtr StorageFactory::get(
         String destination_database = static_cast<const ASTLiteral &>(*args[0]).value.safeGet<String>();
         String destination_table     = static_cast<const ASTLiteral &>(*args[1]).value.safeGet<String>();
 
-        size_t num_buckets = applyVisitor(FieldVisitorConvertToNumber<size_t>(), typeid_cast<ASTLiteral &>(*args[2]).value);
+        size_t num_buckets = applyVisitor(FieldVisitorConvertToNumber<UInt64>(), typeid_cast<ASTLiteral &>(*args[2]).value);
 
-        time_t min_time = applyVisitor(FieldVisitorConvertToNumber<size_t>(), typeid_cast<ASTLiteral &>(*args[3]).value);
-        time_t max_time = applyVisitor(FieldVisitorConvertToNumber<size_t>(), typeid_cast<ASTLiteral &>(*args[4]).value);
-        size_t min_rows = applyVisitor(FieldVisitorConvertToNumber<size_t>(), typeid_cast<ASTLiteral &>(*args[5]).value);
-        size_t max_rows = applyVisitor(FieldVisitorConvertToNumber<size_t>(), typeid_cast<ASTLiteral &>(*args[6]).value);
-        size_t min_bytes = applyVisitor(FieldVisitorConvertToNumber<size_t>(), typeid_cast<ASTLiteral &>(*args[7]).value);
-        size_t max_bytes = applyVisitor(FieldVisitorConvertToNumber<size_t>(), typeid_cast<ASTLiteral &>(*args[8]).value);
+        time_t min_time = applyVisitor(FieldVisitorConvertToNumber<UInt64>(), typeid_cast<ASTLiteral &>(*args[3]).value);
```

Comment:
This looks weird.

Reply:
What do you suggest? `size_t` aliases `unsigned long` for which there's no Poco numeric type. I left the return values implicitly converted to previously used type, as it's convertible on all your supported platforms.

Reply:
But why exactly FieldVisitorConvertToNumber doesn't work?
It should convert to any type, that support implicit conversion.

If it is because of missing implementation of `TypeName`, then better to fix `FieldVisitorConvertToNumber`.

Reply:
Ok, I can add `unsigned long` to `FieldVisitorConvertToNumber`, let me rework this.

Reply:
It is still not the proper solution.

Reply:
I'm not sure there's a proper solution, proper solution is not to use `unsigned long` or `size_t` as it doesn't have equivalent in `Core/Types.h`. It can be UInt32 or UInt64.

Reply:
Ok.

Reply:
The problem is this https://github.com/pocoproject/poco/commit/8ab8dcc3689adeda2c61f9fac1f54b8e5c1cba34#diff-174dd9b4321b9b398d7349f58aae47d0L96
Previously `unsigned long` was alias of UInt64, now it isn't.

Reply:
I see.
So, why don't replace also
`size_t max_rows`
to
`UInt64 max_rows`?